eap74-rhel8-byos-multivm/src/main/bicep/mainTemplate.bicep

```diff
@@ -0,0 +1,519 @@
+          groupIds: [
+            'file'
+          ]
+        }
+      }
+    ]
+    subnet: {
+      id: resourceId(virtualNetworkResourceGroupName, 'Microsoft.Network/virtualNetworks/subnets', virtualNetworkName, subnetName)
+    }
+  }
+  dependsOn: [
+    eapStorageAccount
+  ]
+}
+
+resource virtualNetworkName_resource 'Microsoft.Network/virtualNetworks@2020-11-01' = if (virtualNetworkNewOrExisting == 'new') {
+  name: virtualNetworkName
+  location: location
+  tags: {
+    QuickstartName: 'JBoss EAP on RHEL (clustered, multi-VM)'
+  }
+  properties: {
+    addressSpace: {
+      addressPrefixes: addressPrefixes
+    }
+    subnets: [
+      {
+        name: subnetName
+        properties: {
+          addressPrefix: subnetPrefix
+        }
+      }
+    ]
+  }
+}
+
+resource nicName 'Microsoft.Network/networkInterfaces@2020-11-01' = [for i in range(0, numberOfInstances): {
+  name: concat(nicName_var, i)
+  location: location
+  tags: {
+    QuickstartName: 'JBoss EAP on RHEL (clustered, multi-VM)'
+  }
+  properties: {
+    ipConfigurations: [
+      {
+        name: 'ipconfig1'
+        properties: {
+          privateIPAllocationMethod: 'Dynamic'
+          subnet: {
+            id: resourceId(virtualNetworkResourceGroupName, 'Microsoft.Network/virtualNetworks/subnets', virtualNetworkName, subnetName)
+          }
+          loadBalancerBackendAddressPools: const_enableLoadBalancer ? [
+            {
+              id: resourceId('Microsoft.Network/loadBalancers/backendAddressPools', loadBalancersName_var, backendPoolName)
+            }
+          ] : json('null')
+        }
+      }
+    ]
+  }
+  dependsOn: [
+    virtualNetworkName_resource
+    loadBalancersName
+  ]
+}]
+
+resource vmName_resource 'Microsoft.Compute/virtualMachines@2021-03-01' = [for i in range(0, numberOfInstances): {
+  name: concat(vmName_var, i)
+  location: location
+  plan: plan
+  tags: {
+    QuickstartName: 'JBoss EAP on RHEL (clustered, multi-VM)'
+  }
+  properties: {
+    availabilitySet: {
+      id: asName_resource.id
+    }
+    hardwareProfile: {
+      vmSize: vmSize
+    }
+    networkProfile: {
+      networkInterfaces: [
+        {
+          id: resourceId('Microsoft.Network/networkInterfaces', concat(nicName_var, i))
+        }
+      ]
+    }
+    osProfile: {
+      computerName: concat(vmName_var, i)
+      adminUsername: adminUsername
+      adminPassword: adminPasswordOrSSHKey
+      linuxConfiguration: ((authenticationType == 'password') ? json('null') : linuxConfiguration)
+    }
+    storageProfile: {
+      osDisk: {
+        createOption: 'FromImage'
+      }
+      imageReference: imageReference
+    }
+    diagnosticsProfile: ((bootDiagnostics == 'on') ? json('{"bootDiagnostics": {"enabled": true,"storageUri": "${reference(resourceId(storageAccountResourceGroupName, 'Microsoft.Storage/storageAccounts/', bootStorageName_var), '2021-06-01').primaryEndpoints.blob}"}}') : json('{"bootDiagnostics": {"enabled": false}}'))
```

Comment:
Is this `reference()` invocation going to cause an arm-ttk failure?

Reply:
No it won't, the failure happens when `reference()`is used for referring deployments or deployments outputs